domain/machine/state/state_test.go

```diff
@@ -796,3 +836,28 @@ func (s *stateSuite) TestSetLXDProfilesEmpty(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(profiles, gc.HasLen, 0)
 }
+
+func (s *stateSuite) TestLXDProfiles(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "", "deadbeef")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Insert 2 lxd profiles.
+	db := s.DB()
+	_, err = db.Exec(`INSERT INTO machine_lxd_profile VALUES
+("deadbeef", "profile1")`)
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = db.Exec(`INSERT INTO machine_lxd_profile VALUES
+("deadbeef", "profile2")`)
+	c.Assert(err, jc.ErrorIsNil)
+
+	profiles, err := s.state.LXDProfiles(context.Background(), "deadbeef")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(profiles, gc.HasLen, 2)
```

Comment:
**todo:** remove hasLen check, I think it is redundant with the SameContent check (which already do it)

Reply:
Fixed in 82a3a302d8.